lib/active_record/connection_adapters/sqlserver/database_statements.rb

```diff
@@ -104,6 +138,21 @@ def execute_procedure(proc_name, *variables)
           name = 'Execute Procedure'
           log(sql, name) do
             case @connection_options[:mode]
+            when :jdbc
+              result = begin
+                @connection.fetch(sql).all
+              rescue => e
```

Comment:
The problem is, that I can't detect if the sql statement returns a result set or not.
#fetch uses jdbcs executeQuery (https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#executeQuery(java.lang.String)) which throws an exception when the statement does not yield a result (like ddl statements, in which case the execute(sql) method should be used).

The statement is executed none the less.